components/gitpod-db/src/typeorm/migration/1684328022688-AuthProviderUniqueHostName.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AuthProviderUniqueHostName1684328022688 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        // delete any duplicates
+        await queryRunner.query(`
+            DELETE FROM d_b_auth_provider_entry
+            WHERE id NOT IN (
+                SELECT id FROM (
```

Comment:
nit: the middle  `SELECT id FROM (...) AS t` seems to be redundant. 
anyways, TIL how to delete duplicates 👍🏻